jsk_pcl_ros/src/extract_indices_nodelet.cpp

```diff
@@ -58,6 +58,19 @@ namespace jsk_pcl_ros
     onInitPostProcess();
   }
 
+  ExtractIndices::~ExtractIndices() {
+    // This fixes the following error on shutdown of the nodelet:
+    // terminate called after throwing an instance of
+    // 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
+    //     what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
+    // Also see https://github.com/ros/ros_comm/issues/720 .
```

Comment:
Overall, this PR looks nice, my conceren is that, if some will create new nodelet in future and they did not know about this problem&fixes, and run into same problem, so I think we should add this comment to all nodes, or impliment this fix to all nodelet implcitly. 
https://github.com/jsk-ros-pkg/jsk_recognition/pull/2681#discussion_r865521568